File: program/program/implement-selection-sort/implement_selection_sort.php
<?php

function selectionSort(&$arr) {
    $n = count($arr);

    for ($i = 0; $i < $n - 1; $i++) {
        $min = $i;

        for ($j = $i + 1; $j < $n; $j++) {
            if ($arr[$j] < $arr[$min]) {
                $min = $j;
            }
        }

        // Swap the minimum element with the current element
        if ($min != $i) {
            $temp = $arr[$min];
            $arr[$min] = $arr[$i];
            $arr[$i] = $temp;
        }
    }
}

// Example usage:
$inputArray = [8, 5, 2, 6, 9, 3, 1, 4, 0, 7];

echo "Input Array: [" . implode(", ", $inputArray) . "]\n";

selectionSort($inputArray);

echo "Sorted Array: [" . implode(", ", $inputArray) . "]\n";

?>

File: program/program/implement-selection-sort/implement_selection_sort.r
selection_sort <- function(arr) {
  n <- length(arr)
  
  for (i in 1:(n - 1)) {
    # Set current element as minimum
    min_index <- i
    
    # Find the minimum element in the unsorted portion
    for (j in (i + 1):n) {
      if (arr[j] < arr[min_index]) {
        min_index <- j
      }
    }
    
    # Swap the found minimum element with the current element
    if (min_index != i) {
      temp <- arr[i]
      arr[i] <- arr[min_index]
      arr[min_index] <- temp
    }
  }
  
  return(arr)
}

# Example usage
input <- c(8, 5, 2, 6, 9, 3, 1, 4, 0, 7)
output <- selection_sort(input)
print(output)
